### PR TITLE
rust: build std

### PR DESCRIPTION
**Pull request by Andarwinux**

### CODE REVIEW

File: cmake/packages_check.cmake [hunk 1]
```diff
@@ -18,9 +19,10 @@ elseif(COMPILER_TOOLCHAIN STREQUAL "clang")
     set(ffmpeg_hardcoded_tables "--enable-hardcoded-tables")
     set(mpv_lto_mode "-Db_lto_mode=thin")
     set(mpv_copy_debug COMMAND ${CMAKE_COMMAND} -E copy <BINARY_DIR>/mpv.pdb ${CMAKE_CURRENT_BINARY_DIR}/mpv-debug/mpv.pdb)
+    set(rust_target "gnullvm")
     if(CLANG_PACKAGES_LTO)
         set(cargo_lto_rustflags "CARGO_PROFILE_RELEASE_LTO=thin
-                                 RUSTFLAGS='-C linker-plugin-lto -C embed-bitcode -C lto=thin'")
+                                 RUSTFLAGS='-C target-cpu=${GCC_ARCH} -C control-flow-guard=yes -C linker-plugin-lto -C embed-bitcode -C lto=thin -Z threads=${CPU_COUNT}'")
```
Comment (shinchiro) [inline cmake/packages_check.cmake]: Why there's duplicate flags? It already defined in cargo.config

Comment (Andarwinux) [reply cmake/packages_check.cmake]: https://github.com/rust-lang/cargo/issues/5376
https://doc.rust-lang.org/cargo/reference/config.html#buildrustflags

**Comment by shinchiro** [comment]
Remove `control-flow-guard=yes`. 

Why?
https://doc.rust-lang.org/beta/unstable-book/compiler-flags/control-flow-guard.html#overhead-of-control-flow-guard

**Comment by shinchiro** [comment]
Well, I still dont like the flags since it can potentially increase binary size for no reason. Remove it

**Comment by shinchiro** [comment]
> There is a bug in cfguard

Thats my concern with these flags when it is introduced. I'll remove the cfguard from compiler flags soon

**Pull request closed without merging**